scripts/download_logs.sh

```diff
@@ -58,5 +59,25 @@ function collect_kube_api_resources() {
     done
 }
 
+function download_capi_logs() {
+    collect_capi_provider_resources
+    NAMESPACE=$(get_pod_namespace cluster-api-provider-agent)
+    ${KUBECTL} get pods -n ${NAMESPACE} -o=custom-columns=NAME:.metadata.name --no-headers | xargs -r -I {} sh -c "${KUBECTL} logs {} -n ${NAMESPACE} --all-containers > ${LOGS_DEST}/k8s_{}.log" || true
+    skipper run ./src/junit_log_parser.py --src "${LOGS_DEST}" --dst "${JUNIT_REPORT_DIR}"
+}
+
+function get_pod_namespace() {
+  ${KUBECTL} get pods --no-headers -A  -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.metadata.namespace}{"\n"}' | grep $1 | awk -F " " '{print $2}'
+}
+
+function collect_capi_provider_resources() {
+    JSONPATH=
+    for CR in "${CAPI_PROVIDER_CRS[@]}"
+    do
+      ${KUBECTL} get ${CR} --all-namespaces -o jsonpath='{range .items[*]}{@.metadata.name}{" -n "}{@.metadata.namespace}{"\n"}{end}' --no-headers |\
```

Comment:
what does this line do?

Reply:
Get the CR from all namespaces.
Unlike the assisted-service CRs (though I'm pretty sure that's not entirely true) the provider CRs get created on different namespaces.
This line get all relevant CRs name and namespaces and pass it on the the `kubectl get -o json ...` command

Reply:
Added a comment in the code

Reply:
hmm, but the code is still unreadable I'm afraid
can you make it in a for loop?

Reply:
Done,
Also updated the existing `collect_kube_api_resources` to use for loop